api/envoy/config/ratelimit/v2/rls.proto

```diff
@@ -27,4 +27,10 @@ message RateLimitServiceConfig {
     // requests.
     envoy.api.v2.core.GrpcService grpc_service = 2;
   }
+
+  // Specifies if Envoy should use the data-plane-api client
+  // :repo:`api/envoy/service/ratelimit/v2/rls.proto` or the legacy 
+  // client :repo:`source/common/ratelimit/ratelimit.proto` when
+  // making requests to the rate limit service.
+  bool use_data_plane_proto = 3;
```

Comment:
I think we should invert this so that data_plane_api is default? `use_legacy_proto`? I would also deprecate this right away and explain what the user is supposed to do.

Reply:
I was struggling to decide the correct direction here. My main objection to having use_legacy_proto is that current configuration (absence of the field) would turn on using the data-plane-api proto by default; meaning that this commit would be breaking.

Reply:
Hmm. OK I guess that's OK as long as we deprecate it right away and are clear what the default will become.

Reply:
Exactly. Ok, I will deprecate the field and add more documentation. Also after this lands, I am going to write and email on the mailing lists that has all of the context from here, and from ratelimit as well.